Sources/ScipioKit/ToolchainEnvironment.swift

```diff
@@ -0,0 +1,19 @@
+import struct Basics.AbsolutePath
+@_spi(SwiftPMInternal) import struct Basics.Environment
+
+public typealias ToolchainEnvironment = [String: String]
+
+extension ToolchainEnvironment {
+    var developerDirPath: String? { self["DEVELOPER_DIR"] }
+    var toolchainBinPath: Basics.AbsolutePath? {
+        if let developerDirPath {
+            return try? AbsolutePath(validating: developerDirPath)
+                .appending(components: "Toolchains", "XcodeDefault.xctoolchain", "usr", "bin")
+        }
+        return nil
+    }
+}
+
+extension ToolchainEnvironment? {
```

Comment:
It's confusing to extend to `Optional`. it's better to use `flatMap`

Reply:
The idea is to shift all the boilerplate from the calling site to the extension. Sure, we can use `map` or `flatMap` on the calling site, but then it would be like this:
```swift
try UserToolchain(
    swiftSDK: try .hostSwiftSDK(
        toolchainEnvironment?.toolchainBinPath,
        environment: toolchainEnvironment.map { Environment($0) } ?? Environment.current
    ),
    environment: toolchainEnvironment.map { Environment($0) } ?? Environment.current
)
```
versus with the optional extension:
```swift
try UserToolchain(
    swiftSDK: try .hostSwiftSDK(
        toolchainEnvironment?.toolchainBinPath,
        environment: toolchainEnvironment.asSwiftPMEnvironment
    ),
    environment: toolchainEnvironment.asSwiftPMEnvironment
)
```
Additionally, in the first case, we would also need to use this clunky import: `@_spi(SwiftPMInternal) import struct Basics.Environment` every time `Environment` is referenced.

or did you mean something else?

Reply:
Thank you for the comment. I haven't considered `_spi.`

I already tweaked my changes, but I'll revert this commit. https://github.com/giginet/Scipio/pull/154/commits/260baa5b99ae537c3179333053bbbccccd9be41c

Sorry for the modification!

Reply:
@giginet I see [what you mean](https://github.com/giginet/Scipio/pull/154/commits/260baa5b99ae537c3179333053bbbccccd9be41c)! Looks good to me 👍

Reply:
Oops, I already reverted the commit 😓 

If you think it's good, I'll apply the commit again and merge this. Thank you!

Reply:
I was mistaken, the import was not needed in your case. Yeah, feel free to apply it again. As a benefit of your approach - it's more obvious that the variable is optional at the calling site.